programs/mango-v4/src/instructions/perp_liq_negative_pnl_or_bankruptcy.rs

```diff
@@ -519,25 +520,33 @@ mod tests {
                 liqee_liq_end_health = liqee_health_cache.health(HealthType::LiquidationEnd);
             }
 
-            let insurance_oracle_ai = setup.insurance_oracle.as_account_info();
-            let settle_oracle_ai = setup.settle_oracle.as_account_info();
-            let perp_oracle_ai = setup.perp_oracle.as_account_info();
-
-            let insurance_price = setup
-                .insurance_bank
-                .data()
-                .oracle_price(&AccountInfoRef::borrow(&insurance_oracle_ai).unwrap(), None)
-                .unwrap();
-            let settle_price = setup
-                .settle_bank
-                .data()
-                .oracle_price(&AccountInfoRef::borrow(&settle_oracle_ai).unwrap(), None)
-                .unwrap();
-            let perp_price = setup
-                .perp_market
-                .data()
-                .oracle_price(&AccountInfoRef::borrow(&perp_oracle_ai).unwrap(), None)
-                .unwrap();
+            let insurance_price = {
+                let insurance_oracle_ai = setup.insurance_oracle.as_account_info();
+                let insurance_oracle_ref = &AccountInfoRef::borrow(&insurance_oracle_ai)?;
```

Comment:
while you're touching this code, prefer as_ref() over as_account_info() because the latter clone()s the AccountInfo, which isn't needed here

Reply:
These are `TestAccount`s but I ensured that any instruction uses .as_ref()